vmhost/hostCore/execution.go

```diff
@@ -348,6 +349,47 @@ func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmO
 	return
 }
 
+func (host *vmHost) addNewBackTransfersFromVMOutput(vmOutput *vmcommon.VMOutput, parent, child []byte) {
+	if vmOutput == nil || vmOutput.ReturnCode != vmcommon.Ok {
+		return
+	}
+	callerOutAcc, ok := vmOutput.OutputAccounts[string(parent)]
+	if !ok {
+		return
+	}
+
+	for _, transfer := range callerOutAcc.OutputTransfers {
+		if !bytes.Equal(transfer.SenderAddress, child) {
+			continue
+		}
+		if transfer.CallType == vm.AsynchronousCallBack {
+			continue
+		}
+
+		if transfer.Value.Cmp(vmhost.Zero) > 0 {
+			if len(transfer.Data) == 0 {
+				host.managedTypesContext.AddValueOnlyBackTransfer(transfer.Value)
+			}
+			continue
+		}
+		function, args, err := host.callArgsParser.ParseData(string(transfer.Data))
+		if err != nil {
+			continue
+		}
+
+		esdtTransfers, err := host.esdtTransferParser.ParseESDTTransfers(child, parent, function, args)
+		if err != nil {
+			continue
+		}
+
+		if esdtTransfers.CallFunction != "" {
+			continue
+		}
+
```

Comment:
Optionally, consider extracting as a separate method, returning only a Boolean to add the transfer or not.

Reply:
done